openshift-storage-libs/openshiftstoragelibs/baseclass.py

```diff
@@ -581,6 +590,264 @@ def create_heketi_volume_with_name_and_wait(
+            timeout (int): Timeout for entire volume count to create.
+            interval (int): Time interval between each interation.
+            skip_cleanup (bool): skip cleanup or not.
+
+        Returns:
+            list: list of heketi volume id's.
+
+        Kwargs:
+            The keys, values in kwargs are:
+                - block : (bool)
+                - clusters : (str)|None
+                - disperse_data : (int)|None
+                - durability : (str)|None
+                - gid : (int)|None
+                - gluster_volume_options : (str)|None
+                - name : (str)|None
+                - persistent_volume : (bool)
+                - persistent_volume_endpoint : (str)|None
+                - persistent_volume_file : (str)|None
+                - redundancy : (int):None
+                - replica : (int)|None
+                - size : (int):None
+                - snapshot-factor : (float)|None
+                - json : (bool)
+                - secret : (str)|None
+                - user : (str)|None
+        """
+        kwargs.pop("json", None)
+        h_vols, async_obj = [], []
+
+        if block:
+            # Fetch BHV list before operation
+            bhv_initial = list(get_block_hosting_volume_list(
+                self.heketi_client_node, self.heketi_server_url).keys())
+
+            # Add cleanup function to clean stale volumes created during test
+            self.addCleanup(
+                self.cleanup_heketi_blockhosting_volumes, bhv_initial)
+
+        # Fetch heketi volume list before operation
+        h_vols_count_initial = len(heketi_volume_list(
+            self.heketi_client_node, self.heketi_server_url,
+            json=True).get('volumes'))
+
+        # Temporary replace g.run with g.async_run in heketi_volume_create func
+        # to be able to run it in background. Also, avoid parsing the output as
+        # it won't be json at that moment. Parse it after reading the async
+        # operation results.
+        def run_async(cmd, hostname, raise_on_error=True):
+            async_op = g.run_async(host=hostname, command=cmd)
+            async_obj.append(async_op)
+            return async_op
+
+        # Waiter for overall volumes
+        _waiter = Waiter(timeout=timeout, interval=interval)
+        while vol_count > 0:
+            # Change batch_amount if pvc_count < batch_amount
+            if vol_count < batch_amount:
+                batch_amount = vol_count
+            for _ in range(batch_amount):
+                with mock.patch.object(
+                        json, 'loads', side_effect=(lambda j: j)):
+                    with mock.patch.object(
+                            command, 'cmd_run', side_effect=run_async):
+                        heketi_volume_create(
+                            self.heketi_client_node, self.heketi_server_url,
+                            size, json=True, **kwargs)
```

Comment:
`**kwargs` doesn't have `block` key as  it is passed as function parameter. Pass `block=block` to heketi_ops function

Reply:
Done